src/gt4py/next/program_processors/runners/dace_iterator/itir_to_tasklet.py

```diff
@@ -667,68 +655,50 @@ def _visit_direct_addressing(self, node: itir.FunCall) -> IteratorExpr:
         assert isinstance(tail[0], itir.OffsetLiteral)
         offset = tail[0].value
         assert isinstance(offset, str)
-        shifted_dim = self.offset_provider[offset].value
-
-        assert isinstance(tail[1], itir.OffsetLiteral)
-        shift_amount = tail[1].value
-        assert isinstance(shift_amount, int)
-
-        args = [ValueExpr(iterator.indices[shifted_dim], dace.int64)]
-        internals = [f"{arg.value.data}_v" for arg in args]
-        expr = f"{internals[0]} + {shift_amount}"
-        shifted_value = self.add_expr_tasklet(
-            list(zip(args, internals)), expr, dace.dtypes.int64, "dir_addr"
-        )[0].value
-
-        shifted_index = {dim: value for dim, value in iterator.indices.items()}
-        shifted_index[shifted_dim] = shifted_value
-
-        return IteratorExpr(iterator.field, shifted_index, iterator.dtype, iterator.dimensions)
 
-    def _visit_indirect_addressing(self, node: itir.FunCall) -> IteratorExpr:
-        shift = node.fun
-        assert isinstance(shift, itir.FunCall)
-        tail, rest = self._split_shift_args(shift.args)
-        if rest:
-            iterator = self.visit(self._make_shift_for_rest(rest, node.args[0]))
+        if isinstance(tail[1], itir.OffsetLiteral):
+            offset_node = self.visit_OffsetLiteral(tail[1])
```

Comment:
The reason **all** `visit_*` methods return a `list[ValueExpr]` is because that seemed like the most sane way of dealing with tuples. It's not so nice, but necessary when translating from ASTs to DAGs while eliminating tuples.

If you made `visit_OffsetLiteral` return a `list[ValueExpr]`, even if the list's size is always one, you could get rid of this `if` statement and the explicit call to the node visitor as `visit(tail[1])[0]` would cover all cases.

Reply:
That makes sense, thank you!